### ISSUE TITLE
角度输出

### ISSUE THREAD

**Issue by tinly00** [lx-cly/YOLOv5_OBB_KLD#14]
作者你好！我想请问一下，想要在验证的时候保存目标框的旋转角度的话，需要在哪里改动呢？感谢！

**Comment by lx-cly** [comment]
1.验证的时候可以保存结果成json格式，即`save-json = True`，里面保存了目标旋转框的四个点，在利用` cv2.minAreaRect`可以求出角度。
2.需要保存角度信息至txt文件：修改函数如下
def save_one_txt(predn, save_conf, shape, file):
    # Save one txt result
    gn = torch.tensor(shape)[[1, 0, 1, 0]]  # normalization gain whwh
    for *xyxy,thate,conf, cls in predn.tolist():
        xywh = (xyxy2xywh(torch.tensor(xyxy).view(1, 4)) / gn).view(-1).tolist()  # normalized xywh
        line = (cls, *xywh,thate,conf) if save_conf else (cls, *xywh,thate)  # label format
        with open(file, 'a') as f:
            f.write(('%g ' * len(line)).rstrip() % line + '\n')
 需要保存的predn，添加角度信息。 
 pred_hbbn = torch.cat((hbboxn, pred_polyn[:, -2:]), dim=1) # (n, [xyxy, conf, cls]) native-space pred
 pred_hbba = torch.cat((hbboxn, pred[:,4:5], pred_polyn[:, -2:]), dim=1)

**Comment by tinly00** [comment]
感谢作者大大，我试一下！

**Comment by tinly00** [comment]
你好，我试了一下，我在def save_one_txt(predn, save_conf, shape, file):里面
按你这样修改了，在line那里加了theta参数（你哪个thate是拼错？）
然后在def run里面，加了
pred_hbba = torch.cat((hbboxn, pred[:,4:5], pred_polyn[:, -2:]), dim=1)
（我用的是hukaixuan的代码改的~）
发现在def save_one_txt里面theta会显示没有定义哎？
可以再请教一下吗？
@lx-cly

**Comment by tinly00** [comment]
像这样：
![1671095432678](https://user-images.githubusercontent.com/93133444/207840073-95e8c15d-e769-4bf9-8923-f8cfaffc2c22.jpg)

![1671101314867](https://user-images.githubusercontent.com/93133444/207840259-e443da59-75a0-472f-a31f-d904e89e5b18.png)

**Comment by tinly00** [comment]
> 你好，我试了一下，我在def save_one_txt(predn, save_conf, shape, file):里面 按你这样修改了，在line那里加了theta参数（你哪个thate是拼错？） 然后在def run里面，加了 pred_hbba = torch.cat((hbboxn, pred[:,4:5], pred_polyn[:, -2:]), dim=1) （我用的是hukaixuan的代码改的~） 发现在def save_one_txt里面theta会显示没有定义哎？ 可以再请教一下吗？ @lx-cly

噢发现是for循环里面没有加入theta，但是我再for循环哪里加了theta后，还是报错了：
![1671108580363](https://user-images.githubusercontent.com/93133444/207863320-73b265f9-41d3-4ac8-8a4c-a21e9c15ff81.png)

**Comment by lx-cly** [comment]
save__one_txt函数再runs中调用的时候，传入第一个的参数应该是pred_hbba 而不是pred_hbbn

**Comment by tinly00** [comment]
> save__one_txt函数再runs中调用的时候，传入第一个的参数应该是pred_hbba 而不是pred_hbbn

好的，可以了，感谢！

**Comment by tinly00** [comment]
@lx-cly , 作者大大你好，请问可以扣扣615897363请教一下问题，学习一下吗？
